src/main/webapp/app/shared/form/title-channel-name/title-channel-name.component.ts

```diff
@@ -14,23 +15,42 @@ export class TitleChannelNameComponent implements AfterViewInit, OnDestroy, OnIn
     @Input() titlePattern: string;
     @Input() hideTitleLabel: boolean;
     @Input() emphasizeLabels = false;
-    @Input() hideChannelName?: boolean;
     @Input() minTitleLength: number;
     @Input() initChannelName = true;
+    hideChannelName = input<boolean>();
+    isEditFieldDisplayedRecord = input<Record<ProgrammingExerciseInputField, boolean>>();
+    alreadyUsedTitles = input<string[]>([]);
 
     @ViewChild('field_title') field_title: NgModel;
-    @ViewChild('field_channel_name') field_channel_name?: NgModel;
+    field_channel_name = viewChild<NgModel>('field_channel_name');
 
     @Output() titleChange = new EventEmitter<string>();
     @Output() channelNameChange = new EventEmitter<string>();
 
+    formValidSignal = signal<boolean>(false);
     formValid: boolean;
     formValidChanges = new Subject();
 
     // subscriptions
     fieldTitleSubscription?: Subscription;
     fieldChannelNameSubscription?: Subscription;
 
+    isChannelFieldDisplayed = computed(() => {
+        return !this.hideChannelName() && (!this.isEditFieldDisplayedRecord() || this.isEditFieldDisplayedRecord()?.channelName);
+    });
+
+    constructor() {
+        effect(() => {
+            if (this.isEditFieldDisplayedRecord()?.channelName) {
+            } // triggers the effect
```

Comment:
Unfortunately the linter does not like it
`Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions`
I think that it is easier to read with directly having it the effect here instead of defining another function that would be called